zombies.js

```diff
@@ -79,135 +94,211 @@
+   * Print the player and item names and a message saying the item was discarded.
+   * Return true for the successful discard.
+   * Note: The splice method can also be used for array element replacement.
+   * https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/splice
+   *
+   * If the item is not in the pack, return a message with the item name saying
+   *   nothing was discarded since the item could not be found.
+   * Return false in this case.
+   *
+   * You should be able to invoke this function on a Player instance.
+   *
+   * @name discardItem
+   * @param {Item/Weapon/Food} item   The item to discard.
+   * @return {boolean} true/false     Whether player was able to remove item from pack.
+   */
+
+  this.discardItem = function(item) {
+    if (this._pack.indexOf(item) !== -1) {
+      this._pack.splice(this._pack.indexOf(item), 1);
+      console.log("Item was discarded");
+      return true;
+    }
+    return false;
+  };
+
+  /**
+   * Player Class Method => equip(itemToEquip)
+   * -----------------------------
+   * Player equips a weapon item.
+   *
+   * Player can only equip Weapon instances.
+   * Player can only equip weapon items from their pack.
+   *
+   * If the player already has a weapon equipped (the equipped property
+   *   is set to an Item), find the itemToEquip in the pack and replace
+   *   it with the currently equipped item.  Then set the equipped property
+   *   to the itemToEquip.
+   * However, if the player doesn't already have a weapon equipped, simply
+   *   equip that item and remove it from the pack.
+   * You should be able to invoke this function on a Player instance.
+   *
+   * @name equip
+   * @param {Weapon} itemToEquip  The weapon item to equip.
+   */
+
+  this.equip = function(itemToEquip) {
+    let isWeapon = itemToEquip instanceof Weapon;
+    let currentWeapon = this.equipped;
+    if (!isWeapon || this._pack.indexOf(itemToEquip) === -1) {
+      return false;
+    }
+    if (currentWeapon) {
+      this._pack.push(currentWeapon);
+    }
+    this._pack.splice(this._pack.indexOf(itemToEquip), 1);
+    this.equipped = itemToEquip;
+  };
+
+  /**
+   * Player Class Method => eat(itemToEat)
+   * -----------------------------
+   * Player eats a food item, restoring their health.
+   *
+   * Player can only eat Food instances.
+   * Player can only eat food items from their pack.
+   *
+   * Remove itemToEat from the pack.
+   * Increase the player's health by the food's energy amount, but do not
+   *   exceed the player's max health.  If exceeded, simply set player's health
+   *   to max health instead.
+   * To access the player's max health, be sure to use Player's getMaxHealth method.
+   * You should be able to invoke this function on a Player instance.
+   *
+   * @name eat
+   * @param {Food} itemToEat  The food item to eat.
+   */
+
+  this.eat = function(itemToEat) {
```

Comment:
This should be place outside the function and set on this classes prototype. Right now, you're not setting it on the prototype but on the object instance.